### PR TITLE
Convert play-mvc-2.4 groovy to java

### PR DESCRIPTION
**Pull request by heyams**
part of https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/7195

### CODE REVIEW

File: instrumentation/play/play-mvc/play-mvc-2.4/javaagent/src/play24Test/java/io/opentelemetry/javaagent/instrumentation/play/v2_4/client/PlayWsClientTest.java [hunk 1]
```diff
@@ -78,6 +78,9 @@ protected void configure(HttpClientTestOptions.Builder optionsBuilder) {
           return attributes;
         });
 
+    optionsBuilder.setTestConnectionFailure(false);
+    optionsBuilder.setSpanEndsAfterType(null);
```
Comment (heyams) [inline instrumentation/play/play-mvc/play-mvc-2.4/javaagent/src/play24Test/java/io/opentelemetry/javaagent/instrumentation/play/v2_4/client/PlayWsClientTest.java:82]: spanEndsAfterHeaderReceived is a little bit over 2 seconds, i..e  `2.0509351`

Comment (laurit) [reply instrumentation/play/play-mvc/play-mvc-2.4/javaagent/src/play24Test/java/io/opentelemetry/javaagent/instrumentation/play/v2_4/client/PlayWsClientTest.java:82]: Try adding `optionsBuilder.spanEndsAfterBody();`

File: instrumentation/play/play-mvc/play-mvc-2.4/javaagent/src/play24Test/java/io/opentelemetry/javaagent/instrumentation/play/v2_4/client/PlayWsClientTest.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+import play.libs.ws.WSClient;
+import play.libs.ws.WSRequest;
+import play.libs.ws.WSResponse;
+
+class PlayWsClientTest extends AbstractHttpClientTest<WSRequest> {
+
+  @RegisterExtension
+  static final InstrumentationExtension testing = HttpClientInstrumentationExtension.forAgent();
+
+  WSClient wsClient;
```
Comment (jaydeluca) [inline instrumentation/play/play-mvc/play-mvc-2.4/javaagent/src/play24Test/java/io/opentelemetry/javaagent/instrumentation/play/v2_4/client/PlayWsClientTest.java:35]:
```suggestion
  private static WSClient wsClient;
```

File: instrumentation/play/play-mvc/play-mvc-2.4/javaagent/src/play24Test/java/io/opentelemetry/javaagent/instrumentation/play/v2_4/client/PlayWsClientTest.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+  @BeforeEach
+  void setUp() {
+    wsClient = WS.newClient(-1);
```
Comment (jaydeluca) [inline instrumentation/play/play-mvc/play-mvc-2.4/javaagent/src/play24Test/java/io/opentelemetry/javaagent/instrumentation/play/v2_4/client/PlayWsClientTest.java:39]: The wsClient field uses `@AutoCleanup` in the groovy test. you could add 

```
  @RegisterExtension
  static final AutoCleanupExtension autoCleanup = AutoCleanupExtension.create();
```

and then 
```suggestion
    wsClient = WS.newClient(-1);
    autoCleanup.deferCleanup(wsClient);
```

File: instrumentation/play/play-mvc/play-mvc-2.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/play/v2_4/client/PlayWsClientTest.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+import play.libs.ws.WSResponse;
+
+// Play 2.6+ uses a separately versioned client that shades the underlying dependency
+// This means our built in instrumentation won't work.
+class PlayWsClientTest extends AbstractHttpClientTest<WSRequest> {
+
+  @RegisterExtension
+  static final InstrumentationExtension testing = HttpClientInstrumentationExtension.forAgent();
+
+  WSClient wsClient;
```
Comment (jaydeluca) [inline instrumentation/play/play-mvc/play-mvc-2.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/play/v2_4/client/PlayWsClientTest.java:31]:
```suggestion
  private static WSClient wsClient;
```

File: instrumentation/play/play-mvc/play-mvc-2.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/play/v2_4/client/PlayWsClientTest.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+  @BeforeEach
+  void setUp() {
+    wsClient = WS.newClient(-1);
```
Comment (jaydeluca) [inline instrumentation/play/play-mvc/play-mvc-2.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/play/v2_4/client/PlayWsClientTest.java:35]: same comment here around adding the cleanup

**Review by jaydeluca — APPROVED** [review]

**Comment by trask** [comment]
done 👍

**Pull request merged**